src/main/java/se/liu/ida/hefquin/engine/wrappers/lpgwrapper/impl/LPG2RDFConfigurationReader.java

```diff
@@ -0,0 +1,107 @@
+
+    protected LPG2RDFConfigurationReader(){}
+
+    public static LPG2RDFConfiguration readFromFile(final String filename ) {
+        return instance.parseLPG2RDFConf(filename);
+    }
+
+    public static LPG2RDFConfiguration readFromModel(final Model lpg2Rdf ) {
+        return instance.parseLPG2RDFConf(lpg2Rdf);
+    }
+
+    public LPG2RDFConfiguration parseLPG2RDFConf( final String filename ) {
+        final Model lpg2Rdf = RDFDataMgr.loadModel(filename);
+        return parseLPG2RDFConf(lpg2Rdf);
+    }
+
+    public LPG2RDFConfiguration parseLPG2RDFConf(final Model lpg2Rdf ) {
+
+        final ResIterator lpg2rdfConfigs = lpg2Rdf.listResourcesWithProperty(RDF.type, LPG2RDF.LPGtoRDFConfiguration);
+
+        if (!lpg2rdfConfigs.hasNext()){
+            throw new IllegalArgumentException("LPGtoRDFConfiguration is required!");
+        }
+        final Resource lpg2rdfConfig = lpg2rdfConfigs.next();
+        if(lpg2rdfConfigs.hasNext()){
+            throw new IllegalArgumentException("More than one instance of LPGtoRDFConfiguration!");
+        }
+
+        final String validDataType = lpg2Rdf.getNsPrefixURI("xsd").concat("anyURI");
+
+        final Node label = getLabelPredicate(lpg2rdfConfig, validDataType);
+        final NodeMapping nodeMapping = getNodeMapping(lpg2Rdf, lpg2rdfConfig);
+
+
+        return new LPG2RDFConfigurationImpl(label, nodeMapping);
+    }
+
+    public Node getLabelPredicate(final Resource lpg2rdfConfig, final String validDataType){
+        if(!lpg2rdfConfig.hasProperty(LPG2RDF.labelPredicate)){
+            throw new IllegalArgumentException("labelPredicate is required!");
+        }
+
+        final StmtIterator labelPredicateIterator = lpg2rdfConfig.listProperties(LPG2RDF.labelPredicate);
+
+        if(!labelPredicateIterator.hasNext()){
+            throw new IllegalArgumentException("labelPredicate is required!");
+        }
+        final RDFNode labelPredicateObj = labelPredicateIterator.next().getObject();
+        if(labelPredicateIterator.hasNext()){
+            throw new IllegalArgumentException("More than one labelPredicate!");
+        }
+        if (!labelPredicateObj.asLiteral().getDatatypeURI().equals(validDataType)){
+            throw new IllegalArgumentException("labelPredicate is invalid!");
+        }
+        final String label = labelPredicateObj.asLiteral().getString();
+
+        return NodeFactory.createURI(label);
+    }
+
+    public NodeMapping getNodeMapping(final Model lpg2Rdf, final Resource lpg2rdfConfig){
+        final NodeMapping nodeMapping;
+        if(!lpg2rdfConfig.hasProperty(LPG2RDF.nodeMapping)){
+            throw new IllegalArgumentException("nodeMapping is required!");
+        }
+
+        final Resource nodeMappingResource = lpg2rdfConfig.getProperty(LPG2RDF.nodeMapping).getResource();
+        final RDFNode nodeMappingResourceType = lpg2Rdf.getRequiredProperty(nodeMappingResource, RDF.type).getObject();
```

Comment:
What if `nodeMappingResource` has multiple rdf:type statements? In particular, it may be possible that some RDF description of an LPG-to-RDF configuration describes a node mapping and says that it is of rdf:type `NodeMapping` (the superclass) and of rdf:type `IRIBasedNodeMapping`, or of rdf:type `NodeMapping` and of rdf:type `BNodeBasedNodeMapping`. In this case it may happen that your call of the `getRequiredProperty` function here ends up picking the rdf:type triple with class `NodeMapping` and then you throw the exception below, but the RDF description is not actually wrong.